src/build.d

```diff
@@ -0,0 +1,1105 @@
+	        "-Wno-type-limits",
+	        "-Wno-unknown-pragmas",
+	        "-Wno-unused-function",
+	        "-Wno-unused-label",
+	        "-Wno-unused-parameter",
+	        "-Wno-unused-value",
+	        "-Wno-unused-variable",
+	    ];
+        if (env["CXX_KIND"] == "g++")
+            warnings ~= [
+	            "-Wno-logical-op",
+	            "-Wno-narrowing",
+	            "-Wno-unused-but-set-variable",
+	            "-Wno-uninitialized",
+	            "-Wno-class-memaccess",
+	            "-Wno-implicit-fallthrough",
+	        ];
+        else if (env["CXX_KIND"] == "clang++")
+            warnings ~= "-Wno-logical-op-parentheses";
+    }
+    else
+    {
+        warnings = ["-Wno-deprecated", "-Wstrict-aliasing", "-Werror"];
+    }
+
+    auto targetCPU = "X86";
+    auto cxxFlags = warnings;
+    cxxFlags ~= [
+        "-fno-exceptions", "-fno-rtti",
+        "-D__pascal=", "-DMARS=1", "-DTARGET_" ~ os.toUpper ~ "=1",
+        "-DDM_TARGET_CPU_" ~ targetCPU ~ "=1",
+        env["MODEL_FLAG"],
+        env["PIC_FLAG"],
+    ];
+    if (env["CXX_KIND"] == "g++")
+        cxxFlags ~= ["-std=gnu++98"];
+    if (env["CXX_KIND"] == "clang++")
+        cxxFlags ~= ["-xc++"];
+
+    flags["CXXFLAGS"] = cxxFlags;
+
+    // TODO: allow adding new flags from the environment
+    string[] dflags;
+    flags["DFLAGS"] ~= ["-version=MARS", "-w", "-de", env["PIC_FLAG"], env["MODEL_FLAG"]];
+    // TODO: add different ENABLE modes
+
+    // TODO: remove unused
+    flags["DMD_FLAGS"] = ["-I"~env["D"], "-I"~env["ROOT"], "-Wuninitialized"];
+    flags["GLUE_FLAGS"] = ["-I"~env["D"], "-I"~env["ROOT"], "-I"~env["TK"], "-I"~env["C"]];
+    flags["BACK_FLAGS"] = ["-I"~env["ROOT"], "-I"~env["TK"], "-I"~env["C"], "-I"~env["G"], "-I"~env["D"], "-DDMDV2=1"];
+    flags["ROOT_FLAGS"] = ["-I"~env["ROOT"]];
+
+    // TODO: add support for dObjc
+    auto dObjc = false;
+    version(OSX) version(X86_64)
+        dObjc = true;
+
+    return env;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// D source files
+////////////////////////////////////////////////////////////////////////////////
+
+auto sourceFiles()
+{
+    struct Sources
+    {
+        // TODO: remove unused file lists
+        string[] frontend, lexer, lexerRoot, root, glue, dmd, backend;
+        string[] backendHeaders, tkHeaders, backendC, tkC,
+            backendObjects, opTabGenObjects;
+        string[] glueSrc; // documentation only
+    }
+    string targetCH;
+    string[] targetObjs;
+    if (env["TARGET_CPU"] == "X86")
+    {
+        targetCH = "code_x86.h";
+        targetObjs = ["cg87", "cgxmm", "cgsched", "cod1", "cod2", "cod3", "cod4", "ptrntab"];
+    }
+    else if (env["TARGET_CPU"] == "stub")
+    {
+        targetCH = "code_stub.h";
+        targetObjs = ["platform_stub"];
+    }
+    else
+    {
+        assert(0, "Unknown TARGET_CPU: " ~ env["TARGET_CPU"]);
+    }
+    Sources sources = {
+        frontend: [
+	        "arraytypes",
```

Comment:
Do the source files need to be explicitly enumerated?  I envisioned something more like `sourceDir.dirEntries("*.d", SpanMode.depth).join(" ")`

Reply:
I tried to stay as close as possible to the Makefile, but in general there's no big reason not search for them though  there are a few dead files which are in the codebase, but never used.